nwcsafpps_runner/l1c_processing.py

```diff
@@ -0,0 +1,193 @@
+
+        self.result_home = options.get('output_dir', '/tmp')
+        self.publish_topic = options.get('publish_topic')
+        self.message_data = None
+
+    def initialise(self, service):
+        """Initialise the processor."""
+        self.check_service_name_okay(service)
+        self.l1c_result = None
+        self.pass_start_time = None
+        self.l1cfile = None
+        self.level1_files = []
+        self.service = service
+
+    def run(self, msg):
+        """Start the L1c processing using the relevant sensor specific function from level1c4pps."""
+
+        is_okay = check_message_okay(msg)
+        if not is_okay:
+            return False
+
+        self.platform_name = str(msg.data['platform_name'])
+
+        if not self.check_platform_name_consistent_with_service():
+            return False
+
+        self.sensor = str(msg.data['sensor'])
+        self.message_data = msg.data
+
+        level1_dataset = msg.data['dataset']
+
+        if len(level1_dataset) < 1:
+            return False
+
+        self.get_level1_files_from_dataset(level1_dataset)
+
+        l1c_proc = LVL1C_PROCESSOR_MAPPING.get(self.service)
+        if not l1c_proc:
+            raise AttributeError("Could not find suitable level-1c processor! Service = %s" % self.service)
+
+        self.l1c_result = self.pool.apply_async(l1c_proc, (self.level1_files,
+                                                           self.result_home))
+        return True
+
+    def get_level1_files_from_dataset(self, level1_dataset):
+        """Get the level-1 files from the dataset."""
+
+        if self.service in ['seviri-l1c']:
+            self.level1_files = get_seviri_level1_files_from_dataset(level1_dataset)
+        else:
+            for level1 in level1_dataset:
+                self.level1_files.append(level1['uri'])
+
+    def check_platform_name_consistent_with_service(self):
+        """Check that the platform name is consistent with the service name."""
+
+        if self.platform_name.lower() not in SUPPORTED_SATELLITES.get(self.service, []):
+            LOG.warning("%s: Platform name not supported for this service: %s",
+                        str(self.platform_name), self.service)
+            return False
+
+        return True
+
+
+def get_seviri_level1_files_from_dataset(level1_dataset):
+    """Get the seviri level-1 filenames from the dataset and return as list."""
+
+    pro_files = False
+    epi_files = False
+    level1_files = []
+    for level1 in level1_dataset:
+        level1_filename = level1['uri']
+        level1_files.append(level1_filename)
+        if '-PRO' in level1_filename:
+            pro_files = True
+        if '-EPI' in level1_filename:
+            epi_files = True
+
+    if not pro_files:
+        LOG.warning("PRO file is missing...")
+        return []
+    if not epi_files:
+        LOG.warning("EPI file is missing...")
+        return []
+
+    return level1_files
+
+
+def check_message_okay(msg):
```

Comment:
what does it mean that the message is okay? could this function name be more precise?

Reply:
The thing it checks mor ethan one thing, which indicates I should do more functions. I will think about it...